testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/user/profile/UserProfileTest.java

```diff
@@ -235,6 +235,47 @@ private static void testCustomAttributeInAnyContext(KeycloakSession session) {
         profile.validate();
     }
 
+    @Test
+    public void testEmptyAttributeRemoved() {
+        getTestingClient().server(TEST_REALM_NAME).run((RunOnServer) UserProfileTest::testEmptyAttributeRemoved);
+    }
+
+    private static void testEmptyAttributeRemoved(KeycloakSession session) {
+        Map<String, Object> attributes = new HashMap<>();
+
+        attributes.put(UserModel.USERNAME, org.keycloak.models.utils.KeycloakModelUtils.generateId());
+        attributes.put(UserModel.FIRST_NAME, "John");
+        attributes.put(UserModel.LAST_NAME, "Doe");
+        attributes.put(UserModel.EMAIL, org.keycloak.models.utils.KeycloakModelUtils.generateId() + "@keycloak.org");
+        attributes.put("address", "foo");
+
+        UserProfileProvider provider = getUserProfileProvider(session);
+        UPConfig config = UPConfigUtils.parseSystemDefaultConfig();
+        config.addOrReplaceAttribute(new UPAttribute("address", new UPAttributePermissions(Set.of(), Set.of(ROLE_USER))));
+        provider.setConfiguration(config);
+
+        UserProfile profile = provider.create(UserProfileContext.USER_API, attributes);
+        UserModel user = profile.create();
+
+        // attribute explicitly set with an empty value so we assume it should be removed regardless the `removeAttributes` parameter being set to false
+        attributes.put("address", "");
+        profile = provider.create(UserProfileContext.UPDATE_PROFILE, attributes, user);
+        profile.update(false);
+
+        assertNull(user.getFirstAttribute("address"));
+
+        attributes.put("address", "bar");
+        profile = provider.create(UserProfileContext.UPDATE_PROFILE, attributes, user);
+        profile.update();
+
```

Comment:
What about adding a "pre-condition check" for the next assertion here:

`assertNotNull(user.getFirstAttribute("address"));`

Just to make sure that "bar" has actually been stored.

Reply:
I'll add. We already have similar assertions in other test methods (and in the lines right after)  but we can add that.